[CTOR-438] [Plugin] [CommunityPR] Infor ion grid restapi

lucie-dubrunfaut opened the pull request:
## Description

Adds a new plugin to monitor Infor ION Grid components through its Rest API.
Development from C.Gagnaire [4944](https://github.com/centreon/centreon-plugins/pull/4944)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

cgagnaire commented:
Hi @sc979,
From where do you get this code style ?

sc979 commented:
Hi @cgagnaire ,
Not sure to understand 🤔 
Are you talking about the "Update coding style" commit (915eefb) ?

cgagnaire commented:
Yes I am.

sc979 commented:
This is not coming from the pipeline team.
I will let @lucie-dubrunfaut reply to you

cgagnaire commented:
Sorry, I've mistaken your approvement with the commit, its @lucie-dubrunfaut commit.
Anyway, someone needs to explain to me how this can be seriously considered:
![image](https://github.com/centreon/centreon-plugins/assets/23257354/d3a71bdd-9d94-4bc5-a7cf-7617d9ba31d0)

lucie-dubrunfaut commented:
Hi @cgagnaire , 
We are currently testing a common convergence of coding style in the connectors team. This is still progress and we are open to your feedback regarding this. I was just doing some internal feedback on this one.

cgagnaire commented:
Hi @lucie-dubrunfaut,
No problem, I can say that I don't like it for now :)

lucie-dubrunfaut commented:
I don't really like what's out yet either, but we had to start somewhere, and for now I had to redo a lot of things by hand. :(

cgagnaire commented:
> I don't really like what's out yet either, but we had to start somewhere, and for now I had to redo a lot of things by hand. :(

What are you trying to achieve here ? What's the point of a coding style in this project ?

lucie-dubrunfaut commented on src/apps/infor/ion/grid/api/mode/application.pm:
```diff
@@ -0,0 +1,173 @@
+    $self->{maps_counters_type} = [
+        { name             => 'global',
+          type             => 0,
+          cb_prefix_output => 'prefix_global_output' },
```
Not really it's a kind of mix between it and manual modifications

lucie-dubrunfaut commented:
> > > > I don't really like what's out yet either, but we had to start somewhere, and for now I had to redo a lot of things by hand. :(
> > > 
> > > 
> > > What are you trying to achieve here ? What's the point of a coding style in this project ?
> > 
> > 
> > We try to harmonize the code that is maintained and added here. We used Pycharm's perl coding style parameters but at this moment as we have configured them, they seem not perfectly match. If you have suggestions for tools that would allow us to standardize our coding style, we are open to suggestions.
> 
> Well I don't know about Pycharm and I don't think we need anyone or anything to tell us how we should style our code :) The code I've pushed here https://github.com/centreon/centreon-plugins/pull/4965/files is I think a good example of how we should do. There is no need to be as strict as you did here, this is no PSR.

Okay I thought you were using an automatic code reformatting method to harmonize the coding style of this project. Thank you in any case for your feedback, we can discuss it internally.

On another topic, I added a "total" counter in application mode but I have not access to the API so I can't check myself if it is relevant. If the mode is only used with service discovery and therefore a filter-name, maybe not. But if it is possible to have several applications returned, this counter will potentially be useful for some users. What do you think?

omercier commented on src/apps/infor/ion/grid/custom/restapi.pm:
```diff
@@ -0,0 +1,217 @@
+
+=item B<--hostname>
+
+Set hostname.
+
+=item B<--port>
+
+Set port (default: '443').
+
+=item B<--proto>
+
+Specify https if needed (default: 'https').
+
+=item B<--api-username>
+
+API username.
+
+=item B<--api-password>
+
+API password.
+
+=item B<--timeout>
+
+Threshold for HTTP timeout (default: '20').
```
```suggestion
Define the name or the address of the host.

=item B<--port>

Define the port to connect to (default: '443').

=item B<--proto>

Select the right protocol between 'http' and 'https' (default: 'https').

=item B<--api-username>

Username to access the API.

=item B<--api-password>

Password to access the API.

=item B<--timeout>

Maximum time to wait (in seconds) for a response to HTTP requests before timeout (default: '20').
```

omercier commented on src/apps/infor/ion/grid/mode/application.pm:
```diff
@@ -0,0 +1,181 @@
+
+__END__
+
+=head1 MODE
+
+Check application status.
+
+=over 8
+
+=item B<--filter-name>
+
+Filter by application name (regexp can be used).
+Example: --filter-name='^application1$'
+
+=item B<--warning-status>
+
+Define the conditions to match for the status to be WARNING (default: '').
+Threshold can be matched on special variables like %{state}, %{online}, %{started},
+%{name} or %{description} and Regexp can be used.
+Typical syntax: --warning-status='%{state} ne "OK"'
+
+=item B<--critical-status>
+
+Define the conditions to match for the status to be CRITICAL (default: '%{online} =~ /true/ && %{state} !~ /^(OK)/i').
+Threshold can be matched on special variables like %{state}, %{online}, %{started},
+%{name} or %{description} and Regexp can be used.
+Typical syntax: --critical-status='%{started} ne "true"'
+
+=item B<--warning-total>
+
+Define the conditions to match for the total to be WARNING (default: '').
+
+=item B<--critical-total>
+
+Define the conditions to match for the total to be CRITICAL (default: '').
```
```suggestion
=head1 MODE

Monitor the status of the application.

=over 8

=item B<--filter-name>

Define which applications should be monitored based on their names. This option will be treated as a regular expression.
Example: --filter-name='^application1$'

=item B<--warning-status>

Define the conditions to match for the status to be WARNING (default: '').
The condition can be written using special variables like %{state}, %{online}, %{started},
%{name} or %{description}. Regular expressions are supported.
Typical syntax: --warning-status='%{state} ne "OK"'

=item B<--critical-status>

Define the conditions to match for the status to be CRITICAL (default: '%{online} =~ /true/ && %{state} !~ /^(OK)/i').
The condition can be written using special variables like %{state}, %{online}, %{started},
%{name} or %{description}. Regular expressions are supported.
Typical syntax: --critical-status='%{started} ne "true"'

=item B<--warning-total>

Define the warning threshold for the total number of applications.

=item B<--critical-total>

Define the critical threshold for the total number of applications.
```

omercier commented on src/apps/infor/ion/grid/mode/node.pm:
```diff
@@ -0,0 +1,244 @@
+
+=head1 MODE
+
+Check node status and statistics.
+
+=over 8
+
+=item B<--filter-type>
+
+Filter by node type.
+
+=item B<--filter-name>
+
+Filter by node name.
+
+=item B<--filter-application-name>
+
+Filter by application name.
+
+=item B<--filter-host-name>
+
+Filter by host name.
+
+=item B<--warning-status>
+
+Set warning threshold for status (Default: "%{state} !~ /online/").
+Can use special variables like: %{state}, %{name}, %{host_name},
+%{application_name}, %{type}.
+
+=item B<--critical-status>
+
+Set critical threshold for status.
+Can use special variables like: %{state}, %{name}, %{host_name},
+%{application_name}, %{type}.
+
+=item B<--warning-*> B<--critical-*>
+
+Thresholds.
+Can be: 'log-error', 'log-warning', 'uptime' (s), 'cpu-usage', 'heap-usage' (%).
```
```suggestion
Monitor the status and the statistics of the nodes.

=over 8

=item B<--filter-type>

Define which nodes should be monitored based on the their type. This option will be treated as a regular expression.

=item B<--filter-name>

Define which nodes should be monitored based on the their name. This option will be treated as a regular expression.

=item B<--filter-application-name>

Define which applications should be monitored based on the their name. This option will be treated as a regular expression.

=item B<--filter-host-name>

Define which hosts should be monitored based on the their name. This option will be treated as a regular expression.

=item B<--warning-status>

Define the conditions to match to return a warning status (default: "%{state} !~ /online/").
The condition can be written using the following macros: %{state}, %{name}, %{host_name},
%{application_name}, %{type}.

=item B<--critical-status>

Define the conditions to match to return a critical status.
The condition can be written using the following macros: %{state}, %{name}, %{host_name},
%{application_name}, %{type}.

=item B<--warning-*> B<--critical-*>

Thresholds for 'log-error', 'log-warning', 'uptime' (s), 'cpu-usage', 'heap-usage' (%).
```

cgagnaire commented:
This coding style is not good :)

lucie-dubrunfaut commented:
> This coding style is not good :)

Hi @cgagnaire :)
Can you tell me precisely what you think is wrong so that I can take note of it and we can discuss it again on our side?
